Added Ansible playbook to setup actors dev env

artmello opened the pull request.

artmello committed:
Add Ansible playbook to setup actors dev env

artmello commented:
With this PR I tried to setup a dev environment to someone focusing on actors development. So, for example, snactor is installed via RPM, instead of git clone && make install. After some conversation it seems that maybe this should be focused on someone who wants to develop leapp tools itself. With that in mind, another PR #2 was made with some changes from this one. We probably should go with one or another (or maybe a mix of both)

shaded-enmity commented on ansible/roles/actors/tasks/main.yml:
```diff
@@ -0,0 +1,25 @@
+
+- name: Install packages requested by some actors, if necessary
+  become: true
+  become_method: sudo
```
It is better to set `become_method` in ansible.cfg / playbook - tasks themselves shouldn't need to know how the privilege is being escalated

shaded-enmity commented on ansible/roles/common/tasks/main.yml:
```diff
@@ -0,0 +1,33 @@
+- name: Install python, if necessary
+  become: true
+  become_method: sudo
+  raw: "test -e /usr/bin/python || dnf install -y {{ common.python_pkg }}"
```
I'd probably just include python into `common.packages`

artmello replied:
Without python installed it seems that ansible's dnf module will fail to run on target machine. I could move all common.packages to be installed in a single task using raw module, but I personally prefer to avoid using raw/command module whenever another option is available. We could also make as pre-requisite that machine have python installed, that would help on the other comment you made below (about $HOME variable)

shaded-enmity replied:
DNF (and Yum) is written in Python, so invoking it pre-determines having Python installed and available, but let me try to make sure that there are no Python2/3 problems.

+1 on avoiding raw/command/shell as it sort of defeats the purpose of Ansible :)

artmello replied:
Yes, you are right. I am using an lxc container with fedora 26 image for testing. Vanilla image has dnf and python3 installed, ansible complains about missing "python" binary.

shaded-enmity commented on ansible/roles/common/defaults/main.yml:
```diff
@@ -0,0 +1,8 @@
+---
+common:
+    bashrc: "{{ lookup('env', 'HOME') }}/.bashrc"
```
This will be a problem for people using different shell than Bash (I am Zsh person myself), but I don't think we need to fix this right now, so I'd probably suggest 2 step approach:
1) Document the fact that this playbook currently works only for Bash in a README
2) Add a message to the end of playbook output saying that the env only supports bash

artmello replied:
It seems that after ansible 1.4 I could use ansilbe_env.HOME to get user's home dir (not sure if that handle zsh but I would expect so). But, this variable is only populated after ansible gathering facts from target machine. On this playbook i disabled gathering facts because it seems ansible needs python installed to get such facts (at least that was what I see on my tests). We could:
```
getent passwd {{ user }} | cut -d: -f6
```
or something like it, into an specific task to get it. Or, again, enforce that python is installed on target machine. Which to be honest should be a fair guess on workstation installs, but a possible problem on lxc/docker images

shaded-enmity replied:
Ahh, sorry I didn't exaplain myself :) `$HOME` should be universal, I meant `.bashrc`

artmello replied:
oh, ofc! Sorry about that :) OK, in fact we don't need .bashrc anymore. We were using it just to set up snactor env vars, but after latest snactor update it can use a .sanctor.cfg for that. Will update it

shaded-enmity commented on ansible/roles/snactor/tasks/main.yml:
```diff
@@ -0,0 +1,19 @@
+- name: Enable leapp repository
+  become: true
+  become_method: sudo
+  raw: "dnf copr enable {{ snactor.repository }} -y"
```
nitpick: put `-y` after `enable`

artmello committed:
Add a couple of fixes after review request

- Add go packages required for daemon testing
- Add symlink for leapp-go to working dir
- Add task to download leapp-to/leapp repo

artmello commented:
FWIW, this PR is updated with relevant fixes from #2
